depthmapXcli/runmethods.cpp

```diff
@@ -547,4 +547,77 @@ namespace dm_runmethods
         DO_TIMED("Writing graph", mGraph->write(clp.getOuputFile().c_str(),METAGRAPH_VERSION, false))
                 std::cout << " ok" << std::endl;
     }
+
+    void runMapConversion(const CommandLineParser &clp, const MapConvertParser &mcp, IPerformanceSink &perfWriter)
+    {
+        auto mGraph = loadGraph(clp.getFileName().c_str(), perfWriter);
+
+        int currentMapType = mGraph->getDisplayedMapType();
+
+        if (currentMapType == ShapeMap::EMPTYMAP) {
+            throw depthmapX::RuntimeException("No currently available map to convert from");
+        }
+
+        if (mcp.copyAttributes()) {
+            if(currentMapType == ShapeMap::DATAMAP ||
```

Comment:
Isn't that exactly the wrong way round? You are throwing when the current map *is* one of the allowed types.